### PR TITLE
Fix conflict error while start experiment

### PR DESCRIPTION
**Pull request by YaningX**
Since different docker hosts in the same cloud service can only create one container at the same time,
CloudServiceAvailable is used to check whether a docker host in a cloud service is creating a container.

### CODE REVIEW

File: open-hackathon-server/src/hackathon/hack/host_server_manager.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def get_available_docker_host(self, req_count, hackathon):
         # The new-created VM must run 'cloudvm service by default(either cloud-init or python remote ssh)
         # todo the VM public/private IP will change after reboot, need sync the IP in db with azure in this case
         for docker_host in vms:
-            if self.hosted_docker.ping(docker_host):
+            if self.hosted_docker.ping(docker_host) & self.cloud_service_available.isAvailable(docker_host.public_ip):
```
Comment (juniwang) [inline open-hackathon-server/src/hackathon/hack/host_server_manager.py]: Do you have ever checked the output of `ListHostedService` in azure python sdk? Do we have to lock the object ourself?

File: open-hackathon-server/src/hackathon/hack/host_server_manager.py [hunk 2]
```diff
@@ -681,3 +681,38 @@ def __exist_request_host_server_by_hackathon_id(self, request_count, hackathon_i
...
+    __cloud_service_set = set()
+    __host_lock = thread.allocate_lock()
+
+    def isAvailable(self, public_ip):
+        """
+        check whether a docker host in a cloud service is ready to create a container.
+        :param public_ip: the public ip of a cloud service
+        :return: string such as 123.321.32.3
+        """
+        self.__host_lock.acquire()
```
Comment (juniwang) [inline open-hackathon-server/src/hackathon/hack/host_server_manager.py]: What's happened when a second thread try to acquire the lock and fails? Will it just wait until an lock got or it tries to pick another docker host server without waiting?

**Comment by juniwang** [comment]
Can we go a litter further other than lock? for example, we can combine multiple azure calls into one.

**Comment by YaningX** [comment]
This this pull may be detached, so I create a new pull request.

**Pull request closed without merging**